### PR TITLE
mt7622: remove 300 Hz from dts

### PR DESCRIPTION
**Pull request by graysky2**
Due to the bug described here[1], remove the 300 MHz clock to avoid the low voltage condition that can causes a hang on rebooting RT3200/E8450.

This solution is probably better than the script-based work-around[2].

1. https://forum.openwrt.org/t/belkin-rt3200-linksys-e8450-wifi-ax-discussion/94302/1490
2. https://github.com/openwrt/openwrt/pull/5025

Signed-off-by: John Audia <therealgraysky@proton.me>
Tested-by: Rui Salvaterra <rsalvaterra@gmail.com>
Tested-by: John Audia <therealgraysky@proton.me>

### CODE REVIEW

**Review by MiniSentry — APPROVED** [review]

**Comment by Ansuel** [comment]
@graysky2 again i'm not so happy about this change as someone may perf to run on a broken system and save some power but ok...

I would at least comment the opp instead of totally removing it and adding some comments + some link to the reason it was removed :D

**Comment by jospezial** [comment]
300 Hz-->300 MHz

**Comment by graysky2** [comment]
@Ansuel - good suggestion, done.
@jospezial - thanks, corrected.

File: target/linux/mediatek/patches-5.15/722-remove-300Hz-to-prevent-freeze.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ 	cpu_opp_table: opp-table {
+ 		compatible = "operating-points-v2";
+ 		opp-shared;
+-		opp-300000000 {
+-			opp-hz = /bits/ 64 <300000000>;
+-			opp-microvolt = <950000>;
+-		};
+-
++		
++		/* Due to the bug described here[1], remove the 300 MHz scaling_min)freq to avoid the low
```
Comment (Ansuel) [inline target/linux/mediatek/patches-5.15/722-remove-300Hz-to-prevent-freeze.patch:13]: `scaling_min)freq` ? If you want I can fix it and push this :D

**Comment by Ansuel** [comment]
@hauke can I backport this to 22.03 or is it too late?

**Comment by graysky2** [comment]
Please do

**Comment by hauke** [comment]
@Ansuel  yes you can still backport fixes to 22.03

**Comment by graysky2** [comment]
@rsalvaterra - I have not yet tested this code.  Are you comfortable doing so?

**Comment by graysky2** [comment]
@dangowrt - any thoughts on this?

**Comment by rsalvaterra** [comment]
> @rsalvaterra - I have not yet tested this code. Are you comfortable doing so?

I am, but only when I do a new build (say, when 5.15.55 gets tagged), since it's my production router. I will carry the 300 MHz @ 1 V OPP in my tree, though, since I fathom no physical reason for it to be unsafe (and I'm willing to bet the hardware on it 😉).

**Comment by graysky2** [comment]
Is it better to just increase `opp-microvolt = <950000>;` to `opp-microvolt = <1000000>;` for 300 MHz rather than removing 300 MHz entirely?

**Comment by Ansuel** [comment]
@graysky2 honestly i would remove the 300mhz and backport this to stable... but pls we have to track this and investigate a correct solution... but for now in theory a stable solution is needed to not have broken device in the next release

**Comment by rsalvaterra** [comment]
> @rsalvaterra - Is it better to just increase `opp-microvolt = <950000>;` to `opp-microvolt = <1000000>;` for 300 MHz rather than removing 300 MHz entirely?

That's what I do in my tree: https://github.com/rsalvaterra/openwrt/commit/3f1a0b14b547127a917b54b7a7d08bb00be0adaf

> Perhaps merge this PR and do some testing with the above?

Ack by me on merging this (note that I haven't tested yet, but if it works on your RT3200 it should be fine). Make sure `/sys/devices/system/cpu/cpufreq/policy0/scaling_available_frequencies` doesn't contain `300000`.

>  We can always change it after it is confirmed stable.

As far as stability is concerned, I've been running my RT3200 with the lowest OPP enabled @ 1 V and it never failed on me:
```
root@rocket:~# cat /sys/devices/system/cpu/cpufreq/policy0/scaling_available_frequencies 
300000 437500 600000 812500 1025000 1137500 1262500 1350000 
root@rocket:~# 
```

**Comment by rsalvaterra** [comment]
Looking good on my RT3200.
```
root@rocket:~# cat /sys/devices/system/cpu/cpufreq/policy0/scaling_available_frequencies 
437500 600000 812500 1025000 1137500 1262500 1350000 
root@rocket:~# 
```
@dangowrt, any objections on merging as-is?

**Comment by Ansuel** [comment]
@rsalvaterra in the mean time wants to provide Tested-by tag?

**Comment by graysky2** [comment]
> in the mean time wants to provide Tested-by tag?

Done

**Comment by graysky2** [comment]
I got the same result as expected:
```
# cat /sys/devices/system/cpu/cpufreq/policy0/scaling_available_frequencies 
437500 600000 812500 1025000 1137500 1262500 1350000 
```

**Comment by Ansuel** [comment]
Well guess enough time have passed... stable device have priority..

Merged in master with d0d6b8e1833c587d0c50cac4f6324aa93b0bc8fc
Merged in openwrt-22.03 with 8ccd657629a086d3fb6520a6b91712b1ed0810b7

**Pull request closed without merging**